### DISCUSSION TITLE
Push notification not working in flutter build for ios

### DISCUSSION THREAD

**Discussion by CWThun** [codemagic-ci-cd/codemagic-docs#2338] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

When build app and publish into appstore testflight with XCode, the app run as expected.
But when build with codemagic (same App Id which "push notification" enabled), request token failed from app. 
Below is error received from email
---------------------
We identified one or more issues with a recent delivery for your app, "App Name". Your delivery was successful, but you may wish to correct the following issues in your next delivery:

ITMS-90078: Missing Push Notification Entitlement - Your app appears to register with the Apple Push Notification service, but the app signature's entitlements do not include the 'aps-environment' entitlement. If your app uses the Apple Push Notification service, make sure your App ID is enabled for Push Notification in the Provisioning Portal, and resubmit after signing your app with a Distribution provisioning profile that includes the 'aps-environment' entitlement. Xcode does not automatically copy the aps-environment entitlement from provisioning profiles at build time. This behavior is intentional. To use this entitlement, either enable Push Notifications in the project editor's Capabilities pane, or manually add the entitlement to your entitlements file. For more information, see https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/HandlingRemoteNotifications.html#//apple_ref/doc/uid/TP40008194-CH6-SW1.

### Expected results

can get FCM token from app

### Actual results

[firebase_messaging/unknown]

### Build id (optional)

33784cf7-bb9d-48d5-86b6-79db5a843066

**Comment by himesh-cm** [comment]
Hi @CWThun, Please check if you uploaded the provided profile with APNS enabled.  
Can you please share the build ID with me? You can find it under the 'ID' section in your build details page. Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.

The one you shared is a different id.

**Reply by CWThun** [reply]
Yes, APNS is enabled in the build profile. I use 
⇒build ID: 65b7768bffc50e6991e05a70
My app runs OK when I debug directly from XCode or upload into TestFlight from XCode.
FirebaseMessaging.instance.requestPermission returns "authorized" in both cases.

I use "Automatic" signing method for iOS code sign for building in Codemagic, of course use the same Bundle identifier.

**Reply by himesh-cm** [reply]
Once you add APNS capabilities, please resave the profile and try re-running the build.

**Reply by CWThun** [reply]
I create a new certificate and provisioning profile (Enabled Capabilities: Associated Domains, In-App Purchase, Push Notifications) and re-build, but I got the same error. 
One again, I delete old certificate and provisioning profile and chose [Automatic] iOS code signing method and re-build, but still got the same error. 
What can I do next?

**Reply by himesh-cm** [reply]
Hi @CWThun, Apologies for the delayed reply. The thread was missed. Upon further checking, I found the issue related to code signing. The relevant error is: "exportArchive: No signing certificate 'iOS Distribution' found."
You can follow the guide here - https://docs.codemagic.io/flutter-code-signing/ios-code-signing/

Alternatively, I shared a calendar link to your email id `t***@********.co.jp` for a screen-sharing session.

**Reply by himesh-cm** [reply]
Hi @CWThun.
Is there any update on this?
Can you confirm if you are still facing this issue?

**Comment by mfurkanyuceal** [comment]
Same problem

**Reply by himesh-cm** [reply]
@mfurkanyuceal If you still see this issue, please open a new discussion thread with your build information.

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
